internal/engine/compiler/engine_test.go

```diff
@@ -378,6 +378,7 @@ func ptrAsUint64(f *function) uint64 {
 func TestCallEngine_deferredOnCall(t *testing.T) {
 	vv := &wasm.FunctionType{}
 	s := &wasm.Module{
+		FunctionDefinitionSectionInitialized: 1,
```

Comment:
because we are now using an atomic guard instead of len(), we need to ensure this is set, otherwise it will "reinitialize" in the test, effectively clearing the contents of `FunctionDefinitionSection` for this test (as `buildFunctionDefinitions()` won't find the necessary items to re-fill `FunctionDefinitionSection` correctly).

incidentally, this also requires `FunctionDefinitionSectionInitialized` to be public (eww)

Reply:
sounds like a comment that wants to be in the file ;) maybe tighten up to a one-line why?

Reply:
p.s. any time we make something public for tests, just comment about it. It is icky but also these are internal types.

Reply:
yes, I wanted first to go though your opinion, because this looked super-ugly. I followed @mathetake's suggestion to just use `sync.Once`, then because we no longer have an explicit value to check, I instead added back then `len(m.FunctionDefinitionSection)` at the top of `buildFunctionDefinitions()` because that's the slow path anyway (comment included, open to suggestions anyway)